homeassistant/helpers/update_coordinator.py

```diff
@@ -148,6 +148,8 @@ def _unschedule_refresh(self) -> None:
             self._unsub_refresh()
             self._unsub_refresh = None
 
+        self._debounced_refresh.async_cancel()
```

Comment:
I have simplified the PR, this is now the only code change.

Previously, removing the last listener would unschedule pending refresh but would not cancel the debouncer.